financial-templates-lib/clients/ExpiringMultiPartyEventClient.js

```diff
@@ -8,10 +8,11 @@ class ExpiringMultiPartyEventClient {
    * @param {Object} empAbi Expiring Multi Party truffle ABI object to create a contract instance.
    * @param {Object} web3 Web3 provider from truffle instance.
    * @param {String} empAddress Ethereum address of the EMP contract deployed on the current network.
-   * @param {Integer} latestBlockNumber Offset block number to index events from.
+   * @param {Integer} startingBlockNumber Offset block number to index events from.
+   * @param {Integer} endingBlockNumber Termination block number to index events until. If not defined runs to `latest`.
    * @return None or throws an Error.
    */
-  constructor(logger, empAbi, web3, empAddress, latestBlockNumber = 0) {
+  constructor(logger, empAbi, web3, empAddress, startingBlockNumber = 0, endingBlockNumber = null) {
```

Comment:
Have you checked that this is compatible with the daily reporter? The `GlobalSummaryReporter` uses this module for example

Reply:
The way I've implemented this should not affect any of the existing users of the `ExpiringMultiPartyEventClient`. I have tested the reporter and the normal bot operation and they both perform as expected without any issues :)